src/db.rs

```diff
@@ -6,6 +6,16 @@ use std::collections::{BTreeMap, HashMap};
 use std::sync::{Arc, Mutex};
 use tracing::debug;
 
+/// A wrapper around a `Db` instance. This exists to allow orderly cleanup
+/// of the `Db` by signalling the background purge task to shutdown when
+/// this struct is dropped.
+#[derive(Debug)]
+pub(crate) struct DbHolder {
+    /// The `Db` instance that will be shutdown when this `DbHolder` struct
+    /// is dropped.
```

Comment:
Shutdown without the space is a noun.
```suggestion
    /// The `Db` instance that will be shut down when this `DbHolder` struct
    /// is dropped.
```